### PR TITLE
Fix input typeahead loses focus

### PR DESCRIPTION
**Pull request by brangi**
## Description
Issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/487

## Testing done
Locally.

## Acceptance criteria
- [x] As a keyboard user, I want the Service Type typeahead to retain focus until I have picked a list item. I should be able to tab into the field, pause, and start typing without focus changing.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/facility-locator/components/ServiceTypeAhead.jsx
Comment (kelsonic) [inline src/applications/facility-locator/components/ServiceTypeAhead.jsx]: I'm not super familiar with this code and what `Downshift` is doing with the `key` prop, but do we need a `key` prop here at all? Usually that's only needed for array items within JSX :)

Comment (brangi) [reply src/applications/facility-locator/components/ServiceTypeAhead.jsx]: I do not think we need it , removing it will do the fix too, should we just delete the key here?

Comment (bkjohnson) [reply src/applications/facility-locator/components/ServiceTypeAhead.jsx]: Is the concern that `defaultSelectedItem` can be null?

Comment (kelsonic) [reply src/applications/facility-locator/components/ServiceTypeAhead.jsx]: That makes sense to me! I'd just confirm that it's not used in `Downshift` 👍Well done!

**Review by bkjohnson — COMMENTED** [review]
It isn't clear to me how this solves the problem - more detail would be appreciated.

Also, there is no need to specifically request my review - if you tag the frontend-review-group one of us will get to it.

File: src/applications/facility-locator/components/ServiceTypeAhead.jsx
Comment (brangi) [reply src/applications/facility-locator/components/ServiceTypeAhead.jsx]: @bkjohnson  yes

**Comment by brangi** [comment]
> 
> Also, there is no need to specifically request my review - if you tag the frontend-review-group one of us will get to it.

I don't remember tagging you , maybe it requests your review because you are the CODEOWNER?

**Comment by brangi** [comment]
> It isn't clear to me how this solves the problem - more detail would be appreciated.
> 
@bkjohnson - The issue is that there is an async call happening when the ServiceTypeHead component is mounted that  updates the state so if the view recreates and because `defaultSelectedItem` can be null, the input loses focus. So my fix puts a default value in the key, otherwise I do not see the key being used in the component so we could remove it.  Here is a similar issue: https://stackoverflow.com/questions/22573494/react-js-input-losing-focus-when-rerendering

**Review by bkjohnson — COMMENTED** [review]
I'll wait for peer approval before giving this the frontend approval.

From a platform perspective this looks ok.  However, I agree with @kelsonic that removing the `key` should be considered if it solves the problem and doesn't have any side effects.

**Review by bradl3yC — APPROVED** [review]

**Review by bkjohnson — APPROVED** [review]

**Pull request merged**
